pilot/pkg/xds/ads.go

```diff
@@ -522,6 +523,9 @@ func (s *DiscoveryServer) setProxyState(proxy *model.Proxy, push *model.PushCont
 		return err
 	}
 
+	// this should be done before we look for service instances
+	s.InternalGen.RegisterWorkload(proxy)
```

Comment:
I think this will happen as part of updateProxy on every push, do we want it to trigger this often?

Reply:
Right.. moved it up a level. Can probably move it into OnConnect if I fix the `skipSearchingServiceRegistry` logic.